src/query/functions/src/scalars/map.rs

```diff
@@ -229,6 +240,123 @@ pub fn register(registry: &mut FunctionRegistry) {
+                    return None;
+                }
+            }
+        } else {
+            let key_type = &args_type[1];
+            if !key_type.is_boolean()
+                && !key_type.is_string()
+                && !key_type.is_numeric()
+                && !key_type.is_decimal()
+                && !key_type.is_date_or_date_time()
+            {
+                return None;
+            }
+            for arg_type in args_type.iter().skip(2) {
+                if arg_type != key_type {
+                    return None;
+                }
+            }
+        }
+
+        let return_type = args_type[0].clone();
+
+        Some(Arc::new(Function {
+            signature: FunctionSignature {
+                name: "map_delete".to_string(),
+                args_type: args_type.to_vec(),
+                return_type: return_type.clone(),
+            },
+            eval: FunctionEval::Scalar {
+                calc_domain: Box::new(|_, args_domain| {
+                    FunctionDomain::Domain(args_domain[0].clone())
+                }),
+                eval: Box::new(move |args, _ctx| {
+                    let input_length = args.iter().find_map(|arg| match arg {
+                        ValueRef::Column(col) => Some(col.len()),
+                        _ => None,
+                    });
+
+                    let mut output_map_builder =
+                        ColumnBuilder::with_capacity(&return_type, input_length.unwrap_or(1));
+
+                    for idx in 0..(input_length.unwrap_or(1)) {
+                        let input_map_sref = match &args[0] {
+                            ValueRef::Scalar(map) => map.clone(),
+                            ValueRef::Column(map) => unsafe { map.index_unchecked(idx) },
+                        };
+
+                        match &input_map_sref {
+                            ScalarRef::Null | ScalarRef::EmptyMap => {
+                                output_map_builder.push_default();
+                            }
+                            ScalarRef::Map(col) => {
+                                let mut delete_key_list = HashSet::new();
+
+                                for input_key_item in args.iter().skip(1) {
+                                    let input_key = match &input_key_item {
+                                        ValueRef::Scalar(scalar) => scalar.clone(),
+                                        ValueRef::Column(col) => unsafe {
+                                            col.index_unchecked(idx)
+                                        },
+                                    };
+
+                                    delete_key_list.insert(input_key.to_owned());
+                                }
+
+                                let inner_builder_type = match input_map_sref.infer_data_type() {
+                                    DataType::Map(box typ) => typ,
+                                    DataType::EmptyMap => DataType::EmptyMap,
```

Comment:
this can be removed, as the `ScalarRef::Map(col)` can't have `EmptyMap` type.

Reply:
I have a quick clarification regarding the test cases for the EmptyMap condition. I have captured the current behavior of the function as follows:

```rust
// Deleting keys from an empty map
run_ast(file, "map_delete({})", &[]);

run_ast(file, "map_delete({}, NULL, NULL)", &[]);

// Deleting keys from a map literal
run_ast(file, "map_delete({}, ['k3'], ['k2'])", &[]);
run_ast(
    file,
    "map_delete({'k1': 'v1', 'k2': 'v2', 'k3': 'v3', 'k4': 'v4'}, 'k3', 'k2')",
    &[],
);
```

Here are the unit test results:

```shell
error: 
  --> SQL:1:1
  |
1 | map_delete({})
  | ^^^^^^^^^^^^^^ no overload satisfies `map_delete(Map(Nothing))`


error: 
  --> SQL:1:1
  |
1 | map_delete({}, NULL, NULL)
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^ no overload satisfies `map_delete(Map(Nothing), NULL, NULL)`


error: 
  --> SQL:1:1
  |
1 | map_delete({}, ['k3'], ['k2'])
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ no overload satisfies `map_delete(Map(Nothing), Array(String), Array(String))`
```

Is this behavior aligned with the requirements? I couldn't find details for handling the EmptyMap condition in the documentation: [Snowflake Documentation](https://docs.snowflake.com/en/sql-reference/functions/map_delete).

Reply:
These sqls do not match the type required by the `map_delete` function. The first example has only one argument, and the second and third examples have the wrong type of key value, so return the error is correct.